backend/src/main/java/com/znane_groby/backend/controller/CityController.java

```diff
@@ -0,0 +1,51 @@
+package com.znane_groby.backend.controller;
+
+import com.znane_groby.backend.model.City;
+import com.znane_groby.backend.repository.CityRepository;
+import com.znane_groby.backend.util.ItemNotFoundException;
+import org.springframework.hateoas.CollectionModel;
+import org.springframework.hateoas.EntityModel;
+import org.springframework.hateoas.IanaLinkRelations;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.net.URISyntaxException;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+import static org.springframework.hateoas.server.mvc.WebMvcLinkBuilder.linkTo;
+import static org.springframework.hateoas.server.mvc.WebMvcLinkBuilder.methodOn;
+
+@RestController
+public class CityController {
+    private final CityRepository repository;
+    private final CityResourceAssembler assembler;
+
+    CityController(CityRepository repository, CityResourceAssembler assembler) {
+        this.assembler = assembler;
+        this.repository = repository;
+    }
+
+    // Aggregate root
+    @GetMapping("/cities")
+    CollectionModel<EntityModel<City>> all() {
+
+        List<EntityModel<City>> cities = StreamSupport.stream(repository.findAll().spliterator(), false)
+                .map(assembler::toModel)
+                .collect(Collectors.toList());
+        return new CollectionModel<>(cities,
```

Comment:
O, w sumie śmieszne. Jakoś strasznie dawno temu używałem chyba tego linkowania. Nie wiem czy będzie nam potrzebne, ale całkiem cool że to znalazłeś.